environment/src/test/java/com/sequenceiq/environment/service/integration/CredentialExperienceTest.java

```diff
@@ -302,10 +304,17 @@ private CredentialPrerequisitesResponse getCredentialPrerequisitesResponse(Strin
         @MockBean
         private FlowOperationStatisticsService flowOperationStatisticsService;
 
+        @Bean
+        @Primary
+        MetricService metricService() {
+            return new EnvironmentMetricService();
+        }
```

Comment:
Just for the record, summarizing our offline discussion with Denes: This is needed to avoid the clash with `TestMetricService` in `flow:test`. Due to the `environment/build.gradle` change above, that class is also visible in the `ApplicationContext` besides `EnvironmentMetricService`, so the latter has to be explicitly instantiated as a `@Primary` bean to ensure unique dependency injection.